tests/test_humandetector.py

```diff
@@ -0,0 +1,293 @@
+        """
+        imageWithoutFace = VLImage.load(filename=NO_FACES)
+        detection = self.detector.detect(images=[imageWithoutFace])
+        assert 0 == len(detection[0])
+
+    def test_detect_one_by_area_without_human(self):
+        """
+        Test detection of one human by area without human
+        """
+        detection = self.detector.detectOne(image=VLIMAGE_ONE_FACE, detectArea=AREA_WITHOUT_FACE)
+        assert detection is None, detection
+
+    def test_detect_one_by_area_with_human(self):
+        """
+        Test detection of one human by area with human
+        """
+        detection = self.detector.detectOne(image=VLIMAGE_ONE_FACE, detectArea=GOOD_AREA)
+        self.assertHumanDetection(detection, VLIMAGE_ONE_FACE)
+
+    def test_batch_detect_with_image_of_several_humans(self):
+        """
+        Test batch human detection with image of several humans
+        """
+        detection = self.detector.detect(images=[VLIMAGE_SEVERAL_FACE])
+        self.assertHumanDetection(detection[0], VLIMAGE_SEVERAL_FACE)
+        assert 1 == len(detection)
+        assert 5 == len(detection[0])
+
+    def test_batch_detect_of_multiple_images(self):
+        """
+        Test batch detection of multiple images
+        """
+        detection = self.detector.detect(images=[VLIMAGE_SEVERAL_FACE, VLIMAGE_ONE_FACE])
+        self.assertHumanDetection(detection[0], VLIMAGE_SEVERAL_FACE)
+        self.assertHumanDetection(detection[1], VLIMAGE_ONE_FACE)
+        assert 2 == len(detection)
+        assert 5 == len(detection[0])
+        assert 1 == len(detection[1])
+
+    def test_get_landmarks_for_detect_one(self):
+        """
+        Test get and check landmark instances for detection of one human
+        """
+        for case in self.landmarksCases:
+            with self.subTest(detect=case.detectLandmarks):
+                detection = self.detector.detectOne(image=VLIMAGE_ONE_FACE, detectLandmarks=case.detectLandmarks)
+                self.assertDetectionLandmarks(detection=detection, landmarksIsExpected=case.detectLandmarks)
+
+    def test_get_landmarks_for_batch_detect(self):
+        """
+        Test get and check landmark instances for batch detect
+        """
+        for case in self.landmarksCases:
+            with self.subTest(detect=case.detectLandmarks):
+                detection = self.detector.detect(images=[VLIMAGE_ONE_FACE], detectLandmarks=case.detectLandmarks)[0][0]
+                self.assertDetectionLandmarks(detection=detection, landmarksIsExpected=case.detectLandmarks)
+
+    def test_batch_detect_limit(self):
+        """
+        Test checking detection limit for an image
+        """
+        imageWithManyFaces = VLImage.load(filename=MANY_FACES)
+
+        detection = self.detector.detect(images=[imageWithManyFaces])[0]
+        assert 5 == len(detection)
+
+        detection = self.detector.detect(images=[imageWithManyFaces], limit=20)[0]
+        assert 11 == len(detection)
```

Comment:
limit 20, assert 11?

Reply:
yes.  first i check limit 6, after 20.  Face count is more than 20 on this image.  I check only work of argument limit